### PR TITLE
Implementation of OpenTelemetry metrics transport

### PR DESCRIPTION
**Pull request by h1k3r**
# OpenTelemetry transport implementation

This pull request introduces several enhancements and configurations to support metrics collection and reporting within the OpenTelemetry repository.  

## Changes

1. Changed docker setup to support metrics:  
- Added volume for Grafana data.
- Configured otel-collector to listen on all hosts.
- Added debugging exporter.
- Added Prometheus exporter.
- Configured metrics workflow.
- Configured Prometheus to collect data from otel-collector.
- Changed Tempo volume mapping for updated path in the new image version.
- Added Datadog Export to otel-collector Configuration:  
- Configured Datadog agent.
- Added forwarding of metrics to Datadog agent (commented).
- Added forwarding of metrics directly to Datadog API.

2. Added OpenTelemetry Transport for Metrics:
- Implemented a new transport for handling metrics using OpenTelemetry.
- Added basic configuration to make possible enabling/disabling the transport and setting the metrics namespace.

3. Added OpenTelemetry Metric Tests:
- Added phpunit as a dependency.
- Configured two test suites: unit and integration.
- Added basic tests for metrics transport.

### CODE REVIEW

File: src/OpenTelemetryShopwareBundle.php [hunk 1]
```diff
@@ -27,5 +36,47 @@ public function build(ContainerBuilder $container)
                 ->register('monolog.handler.open_telemetry', Handler::class)
                 ->setFactory([OpenTelemetryLoggerFactory::class, 'build']);
         }
+
+    }
+
+    public function configure(DefinitionConfigurator $definition): void
+    {
+        /** @phpstan-ignore-next-line */
```
Comment (mitelg) [inline src/OpenTelemetryShopwareBundle.php:44]: please put that into the phpstan config with a proper comment, why this is not fixable at the moment. 

just ignoring errors inline is not good

Comment (h1k3r) [reply src/OpenTelemetryShopwareBundle.php]: Fixed as it's done [here](https://github.com/symfony/ux/blob/c69bd672c89f1254ef52fb29f1c93bd78c28f0a3/src/Vue/src/DependencyInjection/VueExtension.php#L80) and in many other symfony bundles

**Review by mitelg — APPROVED** [review]

**Review by mitelg — APPROVED** [review]

File: .github/workflows/cs-fixer.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: CS-Fixer
```
Comment (Gaitholabi) [inline .github/workflows/cs-fixer.yml:1]: I would merge the cs-fixer and php-stan into a single `php-lint.yml` workflow with multiple jobs 🤔

File: phpunit.xml.dist [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    <testsuites>
+        <testsuite name="Unit Test Suite">
+            <directory>tests/unit</directory>
+        </testsuite>
+        <testsuite name="Integration Test Suite">
+            <directory>tests/integration</directory>
+        </testsuite>
+    </testsuites>
+
+    <source ignoreIndirectDeprecations="true" restrictNotices="true" restrictWarnings="true">
```
Comment (Gaitholabi) [inline phpunit.xml.dist:20]: any reason we are restricting notices and warnings?

**Review by Gaitholabi — APPROVED** [review]

File: phpunit.xml.dist (hunk 3 above)
Comment (h1k3r) [reply phpunit.xml.dist:20]: Actually not, it's default config generated by `phpunit --generate-configuration`. Will improve this.

File: .github/workflows/cs-fixer.yml (hunk 2 above)
Comment (h1k3r) [reply .github/workflows/cs-fixer.yml:1]: I would keep it as it is for the moment to be consistent with the other bundle https://github.com/shopware/service-bundle-symfony

**Comment by h1k3r** [comment]
Replaced with https://github.com/shopware/opentelemetry/pull/7

**Pull request closed without merging**